test/TestRist.cpp

```diff
@@ -0,0 +1,280 @@
+        return std::make_shared<RISTNetReceiver::NetworkConnection>();
+    };
+
+    EXPECT_FALSE(receiver.initReceiver(receiverInterfaces, receiverSettings));
+    receiverInterfaces.emplace_back("");
+    EXPECT_FALSE(receiver.initReceiver(receiverInterfaces, receiverSettings));
+    receiverInterfaces.clear();
+    receiverInterfaces.emplace_back("rist://@0.0.0.0:8000");
+    ASSERT_TRUE(receiver.initReceiver(receiverInterfaces, receiverSettings));
+
+    RISTNetSender sender;
+    std::vector<std::tuple<std::string, int>> senderInterfaces;
+    RISTNetSender::RISTNetSenderSettings senderSettings;
+    senderSettings.mPSK = kValidPsk;
+
+    EXPECT_FALSE(sender.initSender(senderInterfaces, senderSettings));
+    senderInterfaces.emplace_back("", 1);
+    EXPECT_FALSE(sender.initSender(senderInterfaces, senderSettings));
+    senderInterfaces.clear();
+    senderInterfaces.emplace_back("rist://127.0.0.1:8000", 1);
+    ASSERT_TRUE(sender.initSender(senderInterfaces, senderSettings));
+}
+
+TEST_F(TestFixture, StartStop) {
+    receiver->getActiveClients(
+        [&](std::map<rist_peer*, std::shared_ptr<RISTNetReceiver::NetworkConnection>>& activeClients) {
+            EXPECT_EQ(activeClients.size(), 1);
+        });
+
+    sender->destroySender();
+    EXPECT_TRUE(checkSenderDisconnecting()) << "Timeout waiting for sender disconnect";
+}
+
+// TODO Enable test when STAR-255 is fixed
+TEST_F(TestFixture, DISABLED_ReceiverCloseConnections) {
+    rist_peer* client = nullptr;
+    receiver->getActiveClients(
+        [&](std::map<rist_peer*, std::shared_ptr<RISTNetReceiver::NetworkConnection>>& activeClients) {
+            EXPECT_EQ(activeClients.size(), 1);
+            client = activeClients.begin()->first;
+        });
+
+    RISTNetSender sender2;
+    ASSERT_TRUE(sender2.initSender(senderInterfaces, senderSettings));
+
+    ASSERT_TRUE(checkSenderConnecting()) << "Timeout waiting for sender2 to connect";
+    receiver->getActiveClients(
+        [&](std::map<rist_peer*, std::shared_ptr<RISTNetReceiver::NetworkConnection>>& activeClients) {
+            EXPECT_EQ(activeClients.size(), 2);
+        });
+
+    EXPECT_TRUE(receiver->closeClientConnection(client));
+    EXPECT_TRUE(checkSenderDisconnecting()) << "Timeout waiting for sender disconnect";
+    receiver->getActiveClients(
+        [&](std::map<rist_peer*, std::shared_ptr<RISTNetReceiver::NetworkConnection>>& activeClients) {
+            EXPECT_EQ(activeClients.size(), 1);
+        });
+
+    receiver->closeAllClientConnections();
+    EXPECT_TRUE(checkSenderDisconnecting()) << "Timeout waiting for sender2 disconnect";
+    receiver->getActiveClients(
+        [&](std::map<rist_peer*, std::shared_ptr<RISTNetReceiver::NetworkConnection>>& activeClients) {
+            EXPECT_EQ(activeClients.size(), 0);
+        });
+}
+
+TEST_F(TestFixture, SendReceive) {
+    uint16_t nReceivedPackets = 0;
```

Comment:
Put this vaiable below the mutex, to make it more clear it is guarded by the mutex :)